workspace_info.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# If a PROW_GIT_TAG variable is provided by Prow's staging image building
+# mechanism, use it instead of asking `git` directly, because the repo we are in
+# might not actually be a Git repo.
+if [[ -n "${PROW_GIT_TAG:-}" ]]; then
+    # Ensure that the PROW_GIT_TAG fits a known pattern of vYYYYMMDD-tag-n-ghash.
+    regex="v[0-9]{8}-[^-]+-[0-9]+-g[0-9a-f]{7}"
+    if ! [[ "${PROW_GIT_TAG}" =~ $regex ]]; then
+        echo >&2 "could not extract git hash from PROW_GIT_TAG ${PROW_GIT_TAG}"
+        exit 1
+    fi
+
+    # Extract the last 7 characters.
+    git_commit="${PROW_GIT_TAG:(-7)}"
+    # Skip built-in date in PROW_GIT_TAG.
+    git_desc="${PROW_GIT_TAG:(10)}"
+    timestamp_utc_rfc3339=$(date -u +"%Y-%m-%d %H:%M:%S%z")
+    timestamp_utc_date_dashes="${timestamp_utc_rfc3339% *}"
+    timestamp_utc_date_no_dashes="${timestamp_utc_date_dashes//-/}"
+else
+    git_commit="$(git rev-parse HEAD)"
+    git_desc_raw="$(git describe --always --dirty --long)"
+    git_desc="${git_desc_raw//\//-}"
+    timestamp_utc_rfc3339=$(date -u +"%Y-%m-%d %H:%M:%S%z")
+    timestamp_utc_date_dashes="${timestamp_utc_rfc3339% *}"
+    timestamp_utc_date_no_dashes="${timestamp_utc_date_dashes//-/}"
+fi
+
+cat << EOF
```

Comment:
You can remove the cat command. Just assign the variables here as-is. That way you don't have to `eval` from the other script.

Reply:
> You can remove the cat command. Just assign the variables here as-is. That way you don't have to `eval` from the other script.

Actually, seeing as how k/k and k/test-infra do the same thing, this is fine as-is.